### PR TITLE
Add band-by-band fluxes for C++ version

### PR DESCRIPTION
**Pull request by brhillman**
Add band-by-band fluxes for C++ version of code. This adds a new class, `FluxesByband`, to the source directory `cpp/extensions/fluxes_byband`, which will keep track of not just the reduced broadband fluxes but also the band by band fluxes. This is needed for integration with E3SM/MMF, which needs band fluxes for other parts of the model.

Note this PR also cleans up the C++ standalone build a little bit.

### CODE REVIEW

File: cpp/rte/mo_rte_lw.h [hunk 1]
```diff
@@ -46,7 +45,92 @@
...
+    #ifdef RRTMGP_EXPENSIVE_CHECKS
+      if (anyLT(inc_flux,0._wp)) { stoprun("rte_lw: inc_flux has values < 0"); }
+    #endif
+  }
+
+  // Number of quadrature points for no-scattering calculation
+  int n_quad_angs = 1;
+  if ( n_gauss_angles != -1 ) {
+    if (n_gauss_angles > max_gauss_pts) {
+      stoprun("rte_lw: asking for too many quadrature points for no-scattering calculation");
```
Comment (mrnorman) [inline cpp/rte/mo_rte_lw.h:94]: Bad user, bad!

File: cpp/test/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# RRTMGPXX testing
+To run tests, execute the following:
+
+    ./get_data.sh
+    ./cmakescript.sh
+    cd build && make && ctest
+
+You will need to set some environment variables before running cmakescript.sh. Examples can be found in macbook.sh.
```
Comment (mrnorman) [inline cpp/test/README.md:8]: Should we add a `source macbook.sh` step to the workflow here?

**Review by mrnorman — APPROVED** [review]
@brhillman have you tested to make sure it still runs on the GPU? If so, I'm happy with this PR.

File: cpp/extensions/fluxes_byband/mo_fluxes_byband_kernels.cpp [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+// Spectral reduction over all points
+void sum_byband(int ncol, int nlev, int ngpt, int nbnd, int2d const &bnd_lims, real3d const &spectral_flux, real3d &byband_flux) {
+  parallel_for( Bounds<3>(nbnd,nlev,ncol) , YAKL_LAMBDA (int ibnd, int ilev, int icol) {
+    real bb_flux_s = 0.0_wp;
+    for (int igpt=bnd_lims(1,ibnd); igpt<=bnd_lims(2,ibnd); igpt++) {
+      bb_flux_s += spectral_flux(icol,ilev,igpt);
+    }
+    byband_flux(icol,ilev,ibnd) = bb_flux_s;
+  });
+  std::cout << "DEBUG: " << byband_flux(1,1,1) << std::endl;
```
[the remark this answers is not in the document]
Comment (brhillman) [reply cpp/extensions/fluxes_byband/mo_fluxes_byband_kernels.cpp]: Yup, removed.

File: cpp/extensions/fluxes_byband/mo_fluxes_byband.h [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        real3d bnd_flux_dn;
+        real3d bnd_flux_dn_dir;
+        real3d bnd_flux_net;
+
+  	void reduce(real3d const &gpt_flux_up, const real3d &gpt_flux_dn, OpticalProps const &spectral_disc,
+                bool top_at_1, real3d const &gpt_flux_dn_dir=real3d()) {
+    	int ncol = size(gpt_flux_up,1);
+    	int nlev = size(gpt_flux_up,2);
+    	int ngpt = size(gpt_flux_up,3);
+        int nbnd = size(this->bnd_flux_up,3);
```
[the remark this answers is not in the document]
Comment (brhillman) [reply cpp/extensions/fluxes_byband/mo_fluxes_byband.h:22]: Mistake. Fixed.

File: cpp/test/README.md (hunk 2 above)
Comment (brhillman) [reply cpp/test/README.md:8]: Nah, I added text describing the workflow for summit instead.

File: cpp/CMakeLists.txt [hunk 5]
```diff
@@ -32,18 +24,12 @@ if ("${ARCH}" STREQUAL "CUDA")
   enable_language(CUDA)
   set_source_files_properties(${CXX_SRC} PROPERTIES LANGUAGE CUDA)
   set_source_files_properties(${CXX_SRC} PROPERTIES COMPILE_FLAGS "${CUDA_FLAGS}")
-  include_directories(${YAKL_CUB_HOME})
+  target_include_directories(rrtmgp PUBLIC ${YAKL_CUB_HOME})
```
[the remark this answers is not in the document]
Comment (brhillman) [reply cpp/CMakeLists.txt]: Removed.

File: cpp/CMakeLists.txt [hunk 6]
```diff
@@ -32,18 +24,12 @@ if ("${ARCH}" STREQUAL "CUDA")
...
 endif()
-include_directories(${YAKL_BIN})
-include_directories(${YAKL_HOME})
-
+target_include_directories(rrtmgp PUBLIC ${YAKL_BIN})
+target_include_directories(rrtmgp PUBLIC ${YAKL_HOME})
```
[the remark this answers is not in the document]
Comment (brhillman) [reply cpp/CMakeLists.txt:29]: Fixed.

**Pull request merged**